exporters/otlp-http/trace/src/main/java/io/opentelemetry/exporter/otlp/http/trace/OtlpHttpSpanExporterBuilder.java

```diff
@@ -127,9 +129,28 @@ public OtlpHttpSpanExporter build() {
       }
     }
```

Comment:
I would consider adding a new method, maybe called `setGrpcEndpoint(String endpoint)`, which expects just an endpoint with scheme, host, and port (like in the otlp grpc exporter builders). If that endpoint is called, `grpc` is enabled and the correct path `GRPC_ENDPOINT_PATH` is automatically appended.

An upside of your approach where grpc is detected based on the endpoint is that this would work out of the box with autoconfigure - no additional properties needed.